python/pyarrow/pandas_compat.py

```diff
@@ -412,14 +412,18 @@ def _get_columns_to_convert_given_schema(df, schema, preserve_index):
                 col = df.index.get_level_values(name)
                 if (preserve_index is None and
                         isinstance(col, _pandas_api.pd.RangeIndex)):
-                    # TODO better error message
-                    raise KeyError("name {} in schema not in columns "
-                                   "or index".format(name))
+                    raise ValueError(
+                        "name '{}' is present in the schema, but it is a "
```

Comment:
```suggestion
                        f"name '{name}' is present in the schema, but it is a "
                        "RangeIndex. Range indices are saved as metadata only- "
                        "they are not converted to a column. Specify 'preserve_index=True' "
                        "to force saving as a column or remove '{name}' from the specified schema."
```

Reply:
That's unfortunately only a recent python feature .. ;) (even when dropping python 2.7, we would not yet be able to use it)